src/entity/projectile/Arrow.php

```diff
@@ -194,7 +194,7 @@ public function onCollideWithPlayer(Player $player) : void{
 			$viewer->getNetworkSession()->onPlayerPickUpItem($player, $this);
 		}
 
-		$playerInventory->addItem(clone $item);
+		$ev->getInventory()->addItem(clone $ev->getItem());
```

Comment:
This should be cloned at the call site.

Also, the inventory might be null, so this needs to be checked before trying to do anything with it.